torchmetrics/functional/text/eed.py

```diff
@@ -0,0 +1,312 @@
+                    row[i - 1] + distance(ref[w - 1], hyp[i - 1]),
+                    row[i] + insertion,
+                )
+            else:
+                nextRow[i] = row[i] + 1.0
+
+        minInd = nextRow.index(min(nextRow))
+        lj[minInd] += 1
+
+        # Long Jumps
+        if ref[w - 1] == " ":
+            jump = alpha + nextRow[minInd]
+            nextRow = [x if x < jump else jump for x in nextRow]
+
+        row = nextRow
+        nextRow = [inf] * (len(hyp) + 1)
+
+    coverage = rho * sum(x if x >= 0 else 1 for x in lj)
+
+    return min(1, (row[-1] + coverage) / (float(len(ref)) + coverage))
+
+
+def preprocess_en(s: str) -> str:
+    """Copied from https://github.com/rwth-i6/ExtendedEditDistance/blob/master/util.py."""
+    if isinstance(s, str) is not True:
+        raise RuntimeError(f"Only strings allowed during preprocessing step, found {type(s)} instead")
+
+    s = s.rstrip()  # trailing space, tab, or newline
+
+    s = s.replace(".", " .")
+    s = s.replace("!", " !")
+    s = s.replace("?", " ?")
+    s = s.replace(",", " ,")
+
+    s = re.sub(r"\s+", r" ", s)  # get rid of extra spaces
+    s = re.sub(r"(\d) ([.,]) (\d)", r"\1\2\3", s)  # 0 . 1 -> 0.1
+    s = re.sub(r"(Dr|Jr|Prof|Rev|Gen|Mr|Mt|Mrs|Ms) .", r"\1.", s)  # Mr . -> Mr.
+    s = s.replace("e . g .", "e.g.")
+    s = s.replace("i . e .", "e.g.")
+    s = s.replace("U . S .", "U.S.")
+    return s
+
+
+def preprocess_ja(s: str) -> str:
+    """Copied from https://github.com/rwth-i6/ExtendedEditDistance/blob/master/util.py."""
+    if isinstance(s, str) is not True:
+        raise RuntimeError(f"Only strings allowed during preprocessing step, found {type(s)} instead")
+
+    s = s.rstrip()  # trailing space, tab, newline
+    s = unicodedata.normalize("NFKC", s)  # まず正規化
+    return s
+
+
+def _eed_compute(scores: tensor, total: tensor) -> tensor:
+    """Final step in extended edit distance.
+
+    Args:
+        scores: sum of individual sentence scores as a tensor
+        total: number of sentences as a tensor
+
+    Returns:
+        average of scores as a tensor
+    """
+    average = scores / total
+    return average
+
+
+def _preprocess_sentences(
+    hypotheses: Union[str, List[str]],
+    references: Union[str, List[str]],
+    language: str,
+) -> Tuple[List[str], List[str]]:
+    """Proprocess strings according to language requirements.
+
+    Args:
+        hypotheses: Transcription(s) to score as a string or list of strings
+        references: Reference(s) for each input as a string or list of strings
+        language: Language used in sentences. Only supports English (en) and Japanese (ja) for now. Defaults to en
+
+    Returns:
+        Tuple of lists that contain the cleaned strings for hypotheses and references
+    """
+    # sanity checks
+    if isinstance(hypotheses, str):
+        hypotheses = [hypotheses]
+    if isinstance(references, str):
+        references = [references]
```

Comment:
If we allow for multiple referenecs, I'd suggest using the method `_validate_inputs` from `torchmetrics.functional.text.helpers.py` instead

Reply:
agree, changed code for next commit